src/index.mjs

```diff
@@ -63,40 +43,87 @@ function isIgnored(node, filter) {
- * @param {Function} options.timeoutFn - Custom timeout function
+ * @param {Object} [options.el] - DOM element to prefetch in-viewport links of
+ * @param {Boolean} [options.priority] - Attempt higher priority fetch (low or high)
+ * @param {Array} [options.origins] - Allowed origins to prefetch (empty allows all)
+ * @param {Array|RegExp|Function} [options.ignores] - Custom filter(s) that run after origin checks
+ * @param {Number} [options.timeout] - Timeout after which prefetching will occur
+ * @param {Number} [options.throttle] - The concurrency limit for prefetching
+ * @param {Number} [options.limit] - The total number of prefetches to allow
+ * @param {Function} [options.timeoutFn] - Custom timeout function
  */
-export default function (options) {
+export function listen(options) {
   if (!options) options = {};
+  if (!window.IntersectionObserver) return;
+
+  const [toAdd, isDone] = throttle(options.throttle || 1/0);
+  const limit = options.limit || 1/0;
 
-  observer && (observer.priority = options.priority || false);
+  // TODO: I think this isn't needed?
+  const isPriority = !!options.priority;
 
   const allowed = options.origins || [location.hostname];
   const ignores = options.ignores || [];
 
-  const timeout = options.timeout || 2e3;
   const timeoutFn = options.timeoutFn || requestIdleCallback;
 
+  const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+      if (entry.isIntersecting) {
+        observer.unobserve(entry = entry.target);
+        // Do not prefetch if will match/exceed limit
+        (++toPrefetch.size >= limit) || toAdd(() => {
+          // TODO: Don't need isPriority?
+          prefetch(entry.href, isPriority).then(isDone);
+        });
+      }
+    });
+  });
+
   timeoutFn(() => {
-    // If URLs are given, prefetch them.
-    if (options.urls) {
-      options.urls.forEach(prefetcher);
-    } else if (observer) {
-      // If not, find all links and use IntersectionObserver.
-      Array.from((options.el || document).querySelectorAll('a'), link => {
-        observer.observe(link);
-        // If the anchor matches a permitted origin
-        // ~> A `[]` or `true` means everything is allowed
-        if (!allowed.length || allowed.includes(link.hostname)) {
-          // If there are any filters, the link must not match any of them
-          isIgnored(link, ignores) || toPrefetch.add(link.href);
-        }
-      });
-    }
-  }, {timeout});
+    // Find all links & Connect them to IO if allowed
+    (options.el || document).querySelectorAll('a').forEach(link => {
+      // If the anchor matches a permitted origin
+      // ~> A `[]` or `true` means everything is allowed
+      if (!allowed.length || allowed.includes(link.hostname)) {
+        // If there are any filters, the link must not match any of them
+        isIgnored(link, ignores) || observer.observe(link);
+      }
+    });
+  }, {
+    timeout: options.timeout || 2e3
```

Comment:
Smallest of nits: could we expand `2e3`? :) I know we're code-golfing but would like to keep the source a wee bit readable for folks who aren't sure why we're using 2000.